source/MaterialXGenGlsl/Nodes/HeightToNormalGlsl.cpp

```diff
@@ -5,237 +5,76 @@
 namespace MaterialX
 {
 
-    SgImplementationPtr HeightToNormalGlsl::create()
-    {
-        return std::make_shared<HeightToNormalGlsl>();
-    }
+HeightToNormalGlsl::HeightToNormalGlsl()
+    : ParentClass()
+{
+    _sampleCount = 9;
+    _sampleSizeFunctionUV.assign("sx_compute_sample_size_uv");
+    _filterFunctionName.assign("sx_normal_from_samples_sobel");
+}
 
-    void HeightToNormalGlsl::createVariables(const SgNode& /*node*/, ShaderGenerator& /*shadergen*/, Shader& shader_)
-    {
-        // Screen size in pixels to be set by client.
-        HwShader& shader = static_cast<HwShader&>(shader_);
-        //const string OGSFX_SIZE_SEMANTIC("ViewportPixelSize"); -- This causes bad code to be emitted.
-        shader.createUniform(HwShader::PIXEL_STAGE, HwShader::PUBLIC_UNIFORMS, Type::VECTOR2, "u_screenSize" /*, OGSFX_SIZE_SEMANTIC*/);
-    }
+SgImplementationPtr HeightToNormalGlsl::create()
+{
+    return std::shared_ptr<HeightToNormalGlsl>(new HeightToNormalGlsl());
+}
 
-    void HeightToNormalGlsl::emitFunctionDefinition(const SgNode& /*node*/, ShaderGenerator& shadergen_, Shader& shader_)
+void HeightToNormalGlsl::computeSampleOffsetStrings(const string& sampleSizeName, StringVec& offsetStrings)
+{
+    // Build a 3x3 grid of samples that are offset by the provided sample size
+    offsetStrings.clear();
+    for (int row = -1; row <= 1; row++)
```

Comment:
e.g. of sample setup. Create a grid of samples 3x3. Box filter for blur is similar but gaussian will have a larger kernel.